### PR TITLE
Upgrade to latest RTIC alpha

### PR DESCRIPTION
**Pull request by hannobraun**
The build was broken due to the release of the new RTIC alpha version.
We pin a specific RTIC version in Cargo.toml to prevent just that, but
that won't prevent Cargo from happily picking up breaking changes from
transitive dependencies. Since RTIC consists of multiple crates, that is
what happened here.

We could explicitly specify and pin all those transitive dependencies in
Cargo.toml to prevent that from happening, but that would get unwieldy.
I decided that it's probably best to just upgrade to the latest alpha,
ride it out for this alpha cycle, and be more careful about adding alpha
dependencies in the future.

### CODE REVIEW

**Comment by hannobraun** [comment]
#311 is blocked by the problem being addressed here. Once this PR is merged, #311 needs a rebase to pick up this change (cc @Lotterleben).

**Comment by Lotterleben** [comment]
ack

**Comment by hannobraun** [comment]
CI fails on nightly only. It seems changes to compiler output cause the compile tests to fail. Stable is green though, so this should be ready to go.

**Review by david-sawatzke — APPROVED** [review]

**Pull request merged**
